From a4f48f7a8ca8980482ff949d2c088a31c8df6b58 Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Tue, 11 Nov 2014 21:18:12 +0800
Subject: [PATCH 0858/5242] MLK-9833-2 dts: imx6sx: do not touch CAN gpios
 pads if M4 is enabled

commit  9277171921f265521445a2d205471d6c1bf90250 from
https://source.codeaurora.org/external/imx/linux-imx.git

CAN devices are allocated to run on M4.
So do not touch CAN pads setting if M4 is enabled.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit 9d2605e51b9ba83382c5da3a838656c9910d75a1)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6sx-sdb-m4.dts |    1 +
 arch/arm/boot/dts/imx6sx-sdb.dtsi   |    9 ++++++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6sx-sdb-m4.dts b/arch/arm/boot/dts/imx6sx-sdb-m4.dts
index 78f1b6b..7b67b8e 100644
--- a/arch/arm/boot/dts/imx6sx-sdb-m4.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb-m4.dts
@@ -17,6 +17,7 @@
 
 &iomuxc {
 	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_hog>;
 };
 
 /*
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dtsi b/arch/arm/boot/dts/imx6sx-sdb.dtsi
index 253684a..d84f5d0 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dtsi
+++ b/arch/arm/boot/dts/imx6sx-sdb.dtsi
@@ -709,7 +709,7 @@
 
 &iomuxc {
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog>;
+	pinctrl-0 = <&pinctrl_hog &pinctrl_can_gpios>;
 
 	imx6x-sdb {
 		pinctrl_hog: hoggrp {
@@ -720,6 +720,13 @@
 			>;
 		};
 
+		pinctrl_can_gpios: can-gpios {
+			fsl,pins = <
+				MX6SX_PAD_QSPI1B_DATA1__GPIO4_IO_25 0x17059
+				MX6SX_PAD_QSPI1B_DATA3__GPIO4_IO_27 0x17059
+			>;
+		};
+
 		pinctrl_audmux: audmuxgrp {
 			fsl,pins = <
 				MX6SX_PAD_CSI_DATA00__AUDMUX_AUD6_TXC	0x130b0
-- 
1.7.9.5

